api/client/network.go

```diff
@@ -46,6 +47,8 @@ func (cli *DockerCli) CmdNetworkCreate(args ...string) error {
 	cmd.Var(flIpamAux, []string{"-aux-address"}, "auxiliary ipv4 or ipv6 addresses used by Network driver")
 	cmd.Var(flOpts, []string{"o", "-opt"}, "set driver specific options")
 
+	flInternal := cmd.Bool([]string{"-internal"}, false, "Configs the network to be internal which disables default gateway service")
```

Comment:
The expected behaviour is that an `internal` network does not have access to the external world.

This is not limited to disabling the default gw service only: Even today we have networks which do not use the default gw service (bridge networks), ultimately they will have to honor the internal flag.